### PR TITLE
[dashboard] Minor Teams & Projects UI fixes

### PR DESCRIPTION
**Pull request by jankeromnes**

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4917#" title="Author self-approved">jankeromnes</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4917#" title="Author self-approved">jankeromnes</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/projects/NewProject.tsx
Comment (jankeromnes) [inline components/dashboard/src/projects/NewProject.tsx:414]: @AlexTugarev Please double-check my suspicion here:
- When creating a new project under a team, we already have a `team` (see `props.onSuccess(team)`)
- Here, we say that the "next" URL after `/reconfigure` should be `/new`

Should it be `/new?team={team.slug}` if there is a `team`?

(I'm trying to fix a weird UX problem where I did "open team page" > "add project" and then got asked "which team?" again, but I don't know what this "next" URL does yet)

Comment (AlexTugarev) [reply components/dashboard/src/projects/NewProject.tsx:414]: @jankeromnes, please ignore `state` param for the time being. It's presence is important for to support the old way marketplace installation _and_ the new project driven installations with the same app. 

> problem where I did "open team page" > "add project" and then got asked "which team?" again

how to reproduce this?

Comment (jankeromnes) [reply components/dashboard/src/projects/NewProject.tsx:414]: > please ignore `state` param for the time being.

Sure, thanks! Will ignore for now.

> how to reproduce this?

It only happened to me once, but then I couldn't reproduce, so I tried to find the bug by looking at the code.

However, since it's hard to reproduce, I think it's fine to ignore it for now. Will remove it from this PR.

File: components/gitpod-db/src/typeorm/team-db-impl.ts
Comment (AlexTugarev) [inline components/gitpod-db/src/typeorm/team-db-impl.ts:78]: 👍🏻 that should work well with URI components

File: components/dashboard/src/components/MonacoEditor.tsx
Comment (AlexTugarev) [inline components/dashboard/src/components/MonacoEditor.tsx:38]: question: would it make sense to move it before L16? is it guaranteed then that the theme is already set before the instance is created?

File: components/gitpod-db/src/typeorm/team-db-impl.ts
Comment (jankeromnes) [reply components/gitpod-db/src/typeorm/team-db-impl.ts:78]: Ah, actually I also wanted to replace them with `-` in the `slug` below (thanks for the reminder). I think keeping `'` in URLs is almost like asking for trouble (even if it's URL-safe).

File: components/dashboard/src/components/MonacoEditor.tsx
Comment (jankeromnes) [reply components/dashboard/src/components/MonacoEditor.tsx:38]: This is independent of the instance (`monaco.editor` vs `editorRef.current`) and acts like a global I think (thus not tied to any particular instance).

Comment (AlexTugarev) [reply components/dashboard/src/components/MonacoEditor.tsx:38]: true. the thing is, if it's already constructed with one, and you change it immediately afterwards, would it render with the "wrong" theme first?

**Comment by gtsiolis** [comment]
@jankeromnes let me know if you need a UX review for this one. I tried going through the project configurator but the GitHub app didn't work. 😭

File: components/dashboard/src/components/MonacoEditor.tsx
Comment (jankeromnes) [reply components/dashboard/src/components/MonacoEditor.tsx:38]: When testing with light and dark themes, it always looked correct / never flashed a wrongly-colored render.

**Review by AlexTugarev — APPROVED** [review]
Changes look good to me.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 736a6afdc29f70c4269d1a310c6e2b5f774c726d</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4917#pullrequestreview-714861359" title="Approved">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4917#" title="Author self-approved">jankeromnes</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [AlexTugarev,jankeromnes]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [AlexTugarev,jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Pull request merged**
